.gitlab-ci.yml

```diff
@@ -116,6 +116,20 @@ test:acceptance:
         fi
 
 
+.publish_helper_functions: &publish_helper_functions |
+  # Bash function to check if the string is a final tag
+  function is_final_tag () {
+    version=$1
+    echo "${version}" | egrep -q '^[0-9]+\.[0-9]+\.[0-9]+$'
+    return $?
+  }
+  # Bash function to check if the string is a build tag
+  function is_build_tag () {
+    version=$1
+    echo "${version}" | egrep -q '^[0-9]+\.[0-9]+\.[0-9]+-build[0-9]+$'
+    return $?
```

Comment:
```suggestion
[[ "${version}" =~ ^[0-9]+\.[0-9]+\.[0-9]+$ ]]
```

Reply:
yeah, part of the argument, `[[` has nothing to do with `set -e`; `[[` is a shell built-in, as opposed to `[` being a command.

Reply:
Ah, I see what you mean now. We could use `[[` and still do `... ]] && return 0 || return $?`.